### PR TITLE
Move dbias from fused attention bwd's input list to its output list

### PR DESCRIPTION
**Pull request by cyanguwa**

### CODE REVIEW

**Comment by zlsh80826** [comment]
LGTM

**Review by zlsh80826 — APPROVED** [review]

File: transformer_engine/pytorch/cpp_extensions.py [hunk 1]
```diff
@@ -188,6 +188,12 @@ def fused_attn_fwd_qkvpacked(
     if attn_scale is None:
         attn_scale = 1.0 / math.sqrt(d)
 
+    if bias_type != "no_bias":
+        assert (bias is not None
+               and bias.shape == [1, h, max_seqlen, max_seqlen]
+               and bias.dtype == qkv.dtype
+               ), """bias tensor cannot be None when bias_type is not no_bias."""
```
Comment (timmoon10) [inline transformer_engine/pytorch/cpp_extensions.py:195]: It'll be easier for the user to debug if we break this up into three one-line asserts.

File: transformer_engine/pytorch/cpp_extensions.py [hunk 2]
```diff
@@ -728,9 +741,9 @@ def fused_attn_bwd_kvpacked(
             d_scale_qkv, d_scale_s, d_scale_o, d_scale_do,
             q_scale_s, q_scale_dp, q_scale_dqkv,
             amax_dp, amax_dqkv,
-            d_bias,
     )
 
+    # returns (d_q, d_kv) when bias_type is no_bias; otherwise returns (d_q, d_kv, d_bias)
     return output_tensors
```
Comment (timmoon10) [inline transformer_engine/pytorch/cpp_extensions.py:747]:
```suggestion
    if bias_type == "no_bias":
        return output_tensors[:2]
    return output_tensors
```

**Review by timmoon10 — CHANGES_REQUESTED** [review]

File: transformer_engine/pytorch/cpp_extensions.py (hunk 1 above)
Comment (cyanguwa) [reply transformer_engine/pytorch/cpp_extensions.py]: Done, thanks!

**Review by ksivaman — APPROVED** [review]

**Pull request merged**